File: dart-algorithms/lib/array_pair.dart



//Algorithm to return a pair array whose sum gives a certain number

  List<List<int>> pairSum(List<int> array, int target){

    Map<int, bool> seenNumbers = {};

    List<List<int>> result = [];

    for(int num in array ){

      int complement = target - num;

      if(seenNumbers.containsKey(complement)){

        result.add([complement, num]);

      }

      seenNumbers[num] = true;

    }

    return result;
  }

File: js-algorithms/src/array_pair.js


function pairSum(array,  target){
    const seenNumbers = new Set();

    const result = [];

    for(let num of array){
        let complement = target - num;

        if(seenNumbers.has(complement)){
            result.push([complement, num]);
        }

        seenNumbers.add(num);
    }

    return result;

}

export default pairSum;